### PR TITLE
REL: backports for yt 4.0.4

### PR DESCRIPTION
**Pull request by neutrinoceros**
## PR Summary

Mainly motivated by #3911.

contents 

- #3891
- #3898
- #3899
- #3912
- #3925
+ version bump and cleanup

Opening as a draft, will open for review when the back ported PRs are actually merged.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -15,8 +15,6 @@
 
 install_ccompiler()
 
-VERSION = "4.0.3"
```
Comment (neutrinoceros) [inline setup.py:18]: this wasn't used anywhere anymore

**Comment by neutrinoceros** [comment]
@matthewturk I took the liberty to setup a small milestone. This PR should contain everything worth backporting since 4.0.3

**Comment by matthewturk** [comment]
Awesome.  I'll look it over tomorrow morning and merge it if it's good.  (Still marked as draft!)

**Comment by neutrinoceros** [comment]
Yup, keeping as a draft to signal that corresponding PRs to main should be dealt with first :)

**Comment by neutrinoceros** [comment]
Keeping as a draft for a couple days more while we wait for a critical upstream fix from pip.

File: setup.py (hunk 1 above)
Comment (matthewturk) [reply setup.py:18]: And I think when we released 4.0.3 we talked about removing it!

**Pull request merged**